### PR TITLE
Deal with explicit NULL dereferences from Coverity

### PR DESCRIPTION
**Pull request by paulidale**
- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: crypto/encode_decode/encoder_pkey.c [hunk 1]
```diff
@@ -261,7 +261,7 @@ static int ossl_encoder_ctx_setup_for_pkey(OSSL_ENCODER_CTX *ctx,
         }
     }
 
-    if (OSSL_ENCODER_CTX_get_num_encoders(ctx) != 0) {
+    if (data != NULL && OSSL_ENCODER_CTX_get_num_encoders(ctx) != 0) {
```
Comment (t8m) [inline crypto/encode_decode/encoder_pkey.c:264]: I suppose this cannot happen as OSSL_ENCODER_CTX_get_num_encoders() would return 0 here in that case. But anyway...

File: providers/implementations/keymgmt/rsa_kmgmt.c [hunk 2]
```diff
@@ -441,7 +441,7 @@ static void *gen_init(void *provctx, int selection, int rsa_type,
             gctx->rsa_type = rsa_type;
         }
     }
-    if (!rsa_gen_set_params(gctx, params)) {
+    if (gctx != NULL && !rsa_gen_set_params(gctx, params)) {
```
Comment (t8m) [inline providers/implementations/keymgmt/rsa_kmgmt.c:444]: Why not just move it inside the if and replace the `gctx = NULL` above with `return NULL`

**Review by t8m — CHANGES_REQUESTED** [review]

File: providers/implementations/keymgmt/rsa_kmgmt.c (hunk 2 above)
Comment (paulidale) [reply providers/implementations/keymgmt/rsa_kmgmt.c]: Done and refactored a little.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master.  Thanks for the approval and comments.

**Pull request closed without merging**